shell/profiling/profiler.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PROFILING_PROFILER_H_
+#define FLUTTER_SHELL_PROFILING_PROFILER_H_
+
+#include <functional>
+#include <memory>
+
+#include "flutter/fml/task_runner.h"
+#include "flutter/fml/trace_event.h"
+
+namespace flutter {
+
+struct CpuUsageInfo {
+  uint32_t num_threads;
+  double total_cpu_usage;
+};
+
+// Return value is the total CPU time spent by all threads owned by the current
+// application. Sometimes the sampler can fail, hence the optional.
+using CpuUsage = std::function<std::optional<CpuUsageInfo>(void)>;
```

Comment:
Rename `CpuUsage` to `CpuUsageGenerator`? My first glance of `CpuUsage` makes me think that it's the same as `CpuUsageInfo`.

Reply:
Moved this to `ProfileSample` struct.